### PR TITLE
Shooting walls can create bullet holes/dents

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
add: Gunfire can now leave bullet holes/dents in walls. Sprites by JStheguy.
/:cl:

Why: I think it looks fun and adds little bits of story around the map as people fight

### CODE REVIEW

**Comment by francinum** [comment]
Is there a way to repair them?

**Comment by KorPhaeron** [comment]
Welding the plate off the wall and replacing it will clear the visuals

**Comment by RemieRichards** [comment]
It should only appear on the dir the projectile came from.

**Comment by optimumtact** [comment]
no offense but they look hideous

**Comment by Mey-Ha-Zah** [comment]
Kor want me to make better bullet holes?

**Comment by francinum** [comment]
So you have to deconstruct the wall to fix it? just make it help-intent with a welder, and anything else deconstruct, like airlocks.

**Comment by optimumtact** [comment]
@francinum it's really not that hard to get the plate off the wall, there's no reason to add another interaction

**Comment by nicbn** [comment]
Make the holes disappear when welding tool is used. Also make them fixed in the x axis for bullets that hit the x axis of the wall and vice-versa for y. That way it won't hit the middle of the wall.

**Comment by Poojawa** [comment]
While it might not be hard to get the plate off the wall, it would be quite annoying to have to spacewalk to grab the plate that promptly gets sucked into space because you're trying to fix walls into space.

if these bullet holes eventually become a thing that causes atmosia reactions, being able to slap like, floor tiles to the wall to patch it up quickly would be useful.

**Comment by Iatots** [comment]
No remie, it should appear on all sides.
That's because you should be able to see through it if you click it while standing close.

**Comment by KorPhaeron** [comment]
@Poojawa 

If they ever become anything but visual we'll definitely add a way to repair it without taking the wall apart

**Comment by DoctoreMad** [comment]
The visuals will make people autistic. Shoot up a gun in maint with a wall facing space and SOMEONE will die. It's even more diabolic than booby trapping paintings because the person that dies to it will know they're going to die.

**Comment by optimumtact** [comment]
true nerd sniping

**Comment by RandomMarine** [comment]
\>OCD drones spacing all of the station's air by fixing bullet holes

**Comment by JStheguy** [comment]
Yeah, I agree with @optimumtact about the effect looking kinda bad, so I had a crack at a bullet hole effect of my own.
image
And the actual sprite
image

File: code/game/turfs/closed.dm [hunk 1]
```diff
@@ -138,3 +138,16 @@
...
+	if(prob(Proj.damage * 2) && !Proj.nodamage && (Proj.damage_type == BRUTE || Proj.damage_type == BURN))
+		var/image/bullet_hole = mutable_appearance('icons/effects/effects.dmi', "bullet_hole", BULLET_HOLE_LAYER)
+
+		var/random_x = rand(-13, 13)
+		bullet_hole.pixel_x += random_x
+
+		var/random_y = rand(-13, 13)
+		bullet_hole.pixel_y += random_y
+
+		add_overlay(bullet_hole)
```
Comment (Cyberboss) [inline code/game/turfs/closed.dm]: make this a priority overlay

File: code/game/turfs/closed.dm [hunk 2]
```diff
@@ -138,3 +138,16 @@
 	desc = "A wall made out of a strange metal. The squares on it pulse in a predictable pattern."
 	icon = 'icons/turf/walls/hierophant_wall.dmi'
 	icon_state = "wall"
+
+/turf/closed/bullet_act(obj/item/projectile/Proj)
+	..()
+	if(prob(Proj.damage * 2) && !Proj.nodamage && (Proj.damage_type == BRUTE || Proj.damage_type == BURN))
+		var/image/bullet_hole = mutable_appearance('icons/effects/effects.dmi', "bullet_hole", BULLET_HOLE_LAYER)
```
Comment (Cyberboss) [inline code/game/turfs/closed.dm]: it's a mutable_appearance, not an image

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/turfs/closed.dm (hunk 2 above)
Comment (RemieRichards) [reply code/game/turfs/closed.dm]: ```/mutable_appearance```s ARE ```/image```s

File: code/game/turfs/closed.dm (hunk 1 above)
Comment (Cyberboss) [reply code/game/turfs/closed.dm]: add_overlay(bullet_hole, TRUE)

**Comment by KorPhaeron** [comment]
@JStheguy thanks

**Comment by nicbn** [comment]
Also can't we check the collision pixel-wise to check where the bullet hits to create the effect?

**Comment by praisenarsie** [comment]
@nicbn we can actually since it exists with target stakes in the test firing areas

**Comment by RemieRichards** [comment]
@praisenarsie not at ALL the same.

**Comment by optimumtact** [comment]
@JStheguy the key this will need to not look bad is variation, like at least 10 or more variants picked randomly at runtime

**Comment by KorPhaeron** [comment]
The key would be better walls honestly. It looks good on r walls because they arent featureless

**Comment by RemieRichards** [comment]
The walls are fine, it IS the bullet holes that are at fault.

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Give it a new sprite then

**Comment by Bawhoppen** [comment]
Energy projectiles should create burn marks.

Also do that thing of bulletholes being a "decal" and only appearing on the side they're from, and make them repairable from a welding tool (ala viscera cleanup detail)

**Comment by Shadowlight213** [comment]
:+1: to making them repairable with a welding tool.
Unless we plan to give the janitor a space suit (please no), requiring the wall to be deconned seems silly.

**Comment by Alek2ander** [comment]
This seems like something the detective should have special interactions with. Scanning to find out the caliber of the bullet, etc.

**Comment by DoctoreMad** [comment]
@Alek2ander Autists CONFIRMED for destroying evidence?

**Comment by Mercenaryblue** [comment]
This is a very neat idea, and it has extremely good potential for the detective. A laser pointer, for example, could reveal exactly where the shots came from.

**Comment by KorPhaeron** [comment]
@Cyberboss

New sprite is looking unlikely

**Pull request merged**